install_config/persistent_storage/persistent_storage_csi.adoc

```diff
@@ -0,0 +1,110 @@
+storage].
+
+
+[NOTE]
+====
+CSI volumes is an alpha feature and may change in a future release of {product-title}. See xref:local-volume-alpha-feature-status[Feature Status(Local Volume)] section for details on known issues and workarounds.
+====
+
+[NOTE]
+====
+{product-title} does not ship any CSI drivers. It is recommended to use CSI drivers provided by link:https://kubernetes-csi.github.io/docs/Drivers.html[community or storage vendors].
+====
+
+[NOTE]
+====
+{product-title} {product-version} supports version 0.2.0 of link:https://github.com/container-storage-interface/spec[CSI specification].
+====
+
+
+[[install-config-persistent-storage-csi-architecture]]
+== Architecture
+
+CSI drivers are typically shipped as container images. These containers are typically not aware of {product-title} where they run. In order to use CSI-compatible storage backed in {product-title}, cluster administrator must deploy several components that sever as a bridge between {product-title} and CSI storage driver.
+
+Following picture provides high-level overview about these components running in pods in {product-title} cluster.
+
+image::csi.png["Architecture of CSI components"]
+
+It is possible to run multiple CSI drivers for different storage backends. Each driver needs its own external attacher StatefulSet, external provisioner Deployment and DaemonSet with the driver and CSI registrar.
+
+=== External CSI attacher StatefulSet
+
+External CSI attacher is a StatefulSet that deploys a single pod with two containers:
+
+  * External CSI attacher container itself that translates `attach` and `detach` calls from {product-title} to respective `ControllerPublish` and `ControllerUnpublish` calls to CSI driver.
+  * CSI driver container.
+
+CSI attacher container talks to CSI driver container using Unix Domain Sockets, ensuring that no CSI communication leaves the pod. CSI driver is not accessible from outside of the pod at all.
+
+StatefulSet is used to ensure that exactly one instance of the pod runs at any given time. External attacher pod does not scale in {product-title} {product-version}.
+
+[NOTE]
+====
+`attach` and `detach` operations typically require CSI driver to use credentials to the storage backend. It is recommended to run external attacher pod runs xref:../../install_config/install/advanced_install.adoc#configuring-dedicated-infrastructure-nodes[infrastructure nodes] so the credentials can never leak to user processes even in case of catastrophic security breach on a node.
+====
+
+[NOTE]
+====
+External attacher must run also for CSI drivers that don't support 3rd party attach/detach operations. External attacher will not issue any `ControllerPublish` or `ControllerUnpublish` to CSI driver, however it still must run to implement necessary {product-title} attachment API.
```

Comment:
s/External attacher/The external attacher

s/must run also/must also run 

s/don't/do not

s/3rd party/third-party

s/`ControllerUnpublish`/`ControllerUnpublish` operations

s/,however/,however,

s/still must run/must still run

s/implement necessary/to implement the necessary